nifi-extension-bundles/nifi-cql-bundle/nifi-cql-services-api/src/main/java/org/apache/nifi/service/cql/api/CQLExecutionService.java

```diff
@@ -0,0 +1,156 @@
+ */
+package org.apache.nifi.service.cql.api;
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.DefaultConsistencyLevel;
+import io.netty.handler.ssl.ClientAuth;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.ControllerService;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.service.cql.api.exception.QueryFailureException;
+import org.apache.nifi.ssl.SSLContextService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public interface CQLExecutionService extends ControllerService {
+    PropertyDescriptor CONTACT_POINTS = new PropertyDescriptor.Builder()
+            .name("Cassandra Contact Points")
+            .description("Contact points are addresses of Cassandra nodes. The list of contact points should be "
+                    + "comma-separated and in hostname:port format. Example node1:port,node2:port,...."
+                    + " The default client port for Cassandra is 9042, but the port(s) must be explicitly specified.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .addValidator(StandardValidators.HOSTNAME_PORT_LIST_VALIDATOR)
+            .build();
+
+    PropertyDescriptor DATACENTER = new PropertyDescriptor.Builder()
+            .name("Cassandra Datacenter")
+            .description("The datacenter setting to use with your node/cluster.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    PropertyDescriptor KEYSPACE = new PropertyDescriptor.Builder()
+            .name("Default Keyspace")
+            .description("The Cassandra Keyspace to connect to. If no keyspace is specified, the query will need to " +
+                    "include the keyspace name before any table reference, in case of 'query' native processors or " +
+                    "if the processor supports the 'Table' property, the keyspace name has to be provided with the " +
+                    "table name in the form of <KEYSPACE>.<TABLE>")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.ENVIRONMENT)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    PropertyDescriptor PROP_SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("SSL Context Service")
+            .description("The SSL Context Service used to provide client certificate information for TLS/SSL "
+                    + "connections.")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    PropertyDescriptor CLIENT_AUTH = new PropertyDescriptor.Builder()
```

Comment:
This property should not be used for client operations, it is only relevant to services that implement listening sockets.